File: python/결혼식(5567).py
from collections import deque

n = int(input())
m = int(input())
graph = [[] for _ in range(n+1)]
for _ in range(m):
    a, b = map(int, input().split()) 
    graph[a].append(b)
    graph[b].append(a)

visited = [0 for _ in range(n+1)]

def bfs(x):
    queue = deque([x])
    visited[x] = 1

    while queue:
        friend_num = queue.popleft()
        for people_num in graph[friend_num]:
            if visited[people_num] == 0:
                queue.append(people_num)
                visited[people_num] = visited[friend_num] + 1


bfs(1)
result = 0
for i in range(2, n+1):
    if 0 < visited[i] <= 3:
        result += 1
print(result)

File: js/결혼식(5567).js
const fs = require('fs');
const input = fs.readFileSync('/dev/stdin').toString().trim().split('\n');

const n = Number(input[0]);
const m = Number(input[1]);
const graph = Array.from({ length: n + 1 }, () => []);

for (let i = 2; i < 2 + m; i++) {
    const [a, b] = input[i].split(' ').map(Number);
    graph[a].push(b);
    graph[b].push(a);
}

const visited = Array(n + 1).fill(0);

function bfs(start) {
    const queue = [];
    queue.push(start);
    visited[start] = 1; 

    while (queue.length > 0) {
        const friendNum = queue.shift(); 

        for (const peopleNum of graph[friendNum]) {
            if (visited[peopleNum] === 0) { 
                queue.push(peopleNum);
                visited[peopleNum] = visited[friendNum] + 1;
            }
        }
    }
}

bfs(1); 

let result = 0;

for (let i = 2; i <= n; i++) {
    if (visited[i] > 0 && visited[i] <= 3) {
        result++;
    }
}

console.log(result);